src/Microsoft.DotNet.Interactive.HttpRequestParser/HttpRequestNode.cs

```diff
@@ -85,11 +88,57 @@ public HttpBindingResult<HttpRequestMessage> TryGetHttpRequestMessage(HttpBindin
         }
         else
         {
-            success = false;
             diagnostics.AddRange(uriBindingResult.Diagnostics);
         }
 
-        if (success)
+        var headers =
+            HeadersNode?.HeaderNodes.Select(h => new KeyValuePair<string, string>(h.NameNode.Text, h.ValueNode.Text)).ToArray()
+            ??
+            Array.Empty<KeyValuePair<string, string>>();
+
+        foreach (var kvp in headers)
+        {
+            switch (kvp.Key.ToLowerInvariant())
+            {
+                case "content-type":
+                    if (request.Content is null)
+                    {
+                        request.Content = new StringContent("");
```

Comment:
Can this be done globally in the `else` clause for the `if` block on line 136 below instead (i.e. always set to empty string if a parsed body is empty)?

The null check for `request.Content` is also confusing since we have not touched `request.Content` above this line. Did you mean to check something else perhaps?

Reply:
No, I meant to check this. Coverage indicates that `request.Content` is null 100% of the time here. But I realized there's a bug in that this header can be get overridden when we set the `Content` again for the body below, so some refactoring is needed.

Reply:
Ah does the System.Net.Http API try to deduce and set the content-type header implicitly when `Content` is set below? That is certainly not very intuitive 😄

I agree this needs some refactoring... I think we can delete this if block for now as it is dead code.

Reply:
The code isn't dead code. 100% of tests enter the `if` block.

> Ah does the System.Net.Http API try to deduce and set the content-type header implicitly when Content is set below?

It doesn't and there's no way it reliably could.